crates/evm/evm/src/executors/invariant/error.rs

```diff
@@ -175,23 +184,36 @@ impl InvariantFuzzError {
             .then_some(CounterExample::Sequence(counterexample_sequence)))
     }
 
-    /// Tests that the modified sequence of calls successfully reverts on the error function.
-    fn fails_successfully<'a>(
+    /// Checks that a subsequence of the provided calls fails the provided invariant test
+    /// and updates an Arc Mutex of the indices of the shortest sequence
+    fn set_fails_successfully(
         &self,
         mut executor: Executor,
-        calls: &'a [BasicTxDetails],
-        anchor: usize,
-        removed_calls: &[usize],
-    ) -> Result<Vec<&'a BasicTxDetails>, ()> {
+        calls: &[BasicTxDetails],
+        use_calls: &[usize],
+        curr_seq: Arc<RwLock<Vec<usize>>>,
+    ) {
+        if curr_seq.read().len() == 1 {
+            // if current sequence is already the smallest possible, just return
+            return;
+        }
```

Comment:
I think we also need to special case when the invariant fails before any calls are executed so sequence length is 0?

Reply:
yeah i thought about this but the difference between 0 and 1 was minor but now that i think about it more, should add the special case (elsewhere)